### PR TITLE
fix: join llc before delegating

### PR DESCRIPTION
**Pull request by guibescos**
Some stake accounts are not created through the createStakeAccount function.
Therefore they need to call joinDaoLlc when delegating/staking.

### CODE REVIEW

File: governance/pyth_staking_sdk/src/pyth-staking-client.ts [hunk 1]
```diff
@@ -810,17 +826,43 @@ export class PythStakingClient {
...
-                isWritable: false,
-                isSigner: false,
-              },
-            ]
+            {
+              pubkey: remainingAccount,
+              isWritable: false,
+              isSigner: false,
+            },
+          ]
```
Comment (guibescos) [inline governance/pyth_staking_sdk/src/pyth-staking-client.ts:834]: my formatter decided to do some stuff

Comment (keyvankhademi) [reply governance/pyth_staking_sdk/src/pyth-staking-client.ts:834]: Does `pnpm fix` allow this?

File: governance/pyth_staking_sdk/src/pyth-staking-client.ts [hunk 2]
```diff
@@ -810,17 +826,43 @@ export class PythStakingClient {
...
           : [],
       )
       .instruction();
   }
 
+  public async getJoinDaoLlcInstructionIfNeeded(
```
Comment (keyvankhademi) [inline governance/pyth_staking_sdk/src/pyth-staking-client.ts:840]: I would make these separated functions.

`hadJoinedDaoLlc` and `getJoinDaoLlcInstruction`

Then you dont have to check if this the result of this function is undefined

**Review by keyvankhademi — APPROVED** [review]

File: governance/pyth_staking_sdk/src/pyth-staking-client.ts (hunk 2 above)
Comment (guibescos) [reply governance/pyth_staking_sdk/src/pyth-staking-client.ts:844]: cool

**Pull request merged**